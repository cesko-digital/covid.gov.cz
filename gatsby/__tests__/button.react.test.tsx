import React from 'react';
import renderer from 'react-test-renderer';

import Button from '@/components/button';
import { Add } from '@material-ui/icons';

jest.mock('@/components/i18n');

test('Button renders HTML button element (not anchor element)', () => {
  const component = renderer.create(
    <>
      <Button variant="contained" />
      <Button href="" variant="contained" />
    </>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button with text renders correctly', () => {
  const component = renderer.create(<Button text="VÍCE" variant="contained" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Anchor with text inside renders correctly', () => {
  const component = renderer.create(
    <Button text="VÍCE" href="#" variant="contained" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button with icon renders correctly', () => {
  const component = renderer.create(
    <Button icon={<Add />} variant="contained" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Buttons with variants render correctly', () => {
  const component = renderer.create(
    <>
      <Button variant="contained" />
      <Button variant="outline" />
      <Button variant="secondary" />
      <Button variant="outline-black" />
      <Button variant="small-black" />
    </>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button does not render unknown variants classes', () => {
  // @ts-ignore
  const component = renderer.create(<Button variant="whatever-button" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button renders with additional class correctly', () => {
  const component = renderer.create(
    <Button className="whatever-button" variant="contained" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Anchors with variants render correctly', () => {
  const component = renderer.create(
    <>
      <Button variant="contained" href="#" />
      <Button variant="outline" href="#" />
      <Button variant="secondary" href="#" />
      <Button variant="outline-black" href="#" />
      <Button variant="small-black" href="#" />
    </>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Anchor does not render unknown variants classes', () => {
  const component = renderer.create(
    // @ts-ignore
    <Button variant="whatever-anchor" href="#" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Anchor renders with additional class correctly', () => {
  const component = renderer.create(
    <Button className="whatever-anchor" href="#" variant="contained" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button with onClick renders correctly', () => {
  const toggleSection = () => {};
  const component = renderer.create(
    <Button onClick={toggleSection} variant="contained" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Anchor element renders correctly', () => {
  const component = renderer.create(<Button href="#" variant="contained" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button with disabled state renders correctly ', () => {
  const component = renderer.create(
    <>
      <Button disabled variant="contained" />
    </>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Anchor with disabled state renders like button ', () => {
  const component = renderer.create(
    <>
      <Button href="#" disabled variant="contained" />
      <Button href="" disabled variant="contained" />
    </>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
