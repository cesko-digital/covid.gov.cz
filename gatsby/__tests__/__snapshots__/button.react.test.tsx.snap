// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor does not render unknown variants classes 1`] = `
<a
  className="btn"
  data-testid="button-link"
  onClick={[Function]}
/>
`;

exports[`Anchor element renders correctly 1`] = `
<a
  className="btn btn-primary"
  data-testid="button-link"
  onClick={[Function]}
/>
`;

exports[`Anchor renders with additional class correctly 1`] = `
<a
  className="btn btn-primary whatever-anchor"
  data-testid="button-link"
  onClick={[Function]}
/>
`;

exports[`Anchor with text inside renders correctly 1`] = `
<a
  className="btn btn-primary"
  data-testid="button-link"
  onClick={[Function]}
>
  VÍCE
</a>
`;

exports[`Anchors with variants render correctly 1`] = `
Array [
  <a
    className="btn btn-primary"
    data-testid="button-link"
    onClick={[Function]}
  />,
  <a
    className="btn btn-outline-primary"
    data-testid="button-link"
    onClick={[Function]}
  />,
  <a
    className="btn btn-secondary"
    data-testid="button-link"
    onClick={[Function]}
  />,
  <a
    className="btn btn-outline-dark text-black"
    data-testid="button-link"
    onClick={[Function]}
  />,
  <a
    className="btnSmallBlack"
    data-testid="button-link"
    onClick={[Function]}
  />,
]
`;

exports[`Button does not render unknown variants classes 1`] = `
<button
  className="btn"
  type="button"
/>
`;

exports[`Button renders HTML button element (not anchor element) 1`] = `
Array [
  <button
    className="btn btn-primary"
    type="button"
  />,
  <a
    className="btn btn-primary"
    data-testid="button-link"
    onClick={[Function]}
  />,
]
`;

exports[`Button renders with additional class correctly 1`] = `
<button
  className="btn btn-primary whatever-button"
  type="button"
/>
`;

exports[`Button with disabled state renders correctly  1`] = `
<button
  className="btn btn-primary"
  disabled={true}
  type="button"
/>
`;

exports[`Button with icon renders correctly 1`] = `
<button
  className="btn btn-primary"
  type="button"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
  </svg>
</button>
`;

exports[`Button with onClick renders correctly 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
/>
`;

exports[`Button with text renders correctly 1`] = `
<button
  className="btn btn-primary"
  type="button"
>
  VÍCE
</button>
`;

exports[`Buttons with variants render correctly 1`] = `
Array [
  <button
    className="btn btn-primary"
    type="button"
  />,
  <button
    className="btn btn-outline-primary"
    type="button"
  />,
  <button
    className="btn btn-secondary"
    type="button"
  />,
  <button
    className="btn btn-outline-dark text-black"
    type="button"
  />,
  <button
    className="btnSmallBlack"
    type="button"
  />,
]
`;
